portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -89,7 +81,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       BlogHashMap blogMap = new BlogHashMap();
       blogMap.putInMap(messages);
 
-      // Load messages from BlogHashMap and respond with gson.
+      // If (user loads all BlogMessages) 
+      LinkedList<BlogMessage> allBlogMessages = blogMap.getMessages();
+
+      if (numberOfCommentsToDisplay < 1 || numberOfCommentsToDisplay >= allBlogMessages.size()) {
+        response.setContentType("text/html");
+        response.getWriter().println("Please enter an integer between 1 and "+allBlogMessages.size()+".");
+        return;
+      }
+
+      // If (user loads BlogMessages for a specific tag)
+      String tagToSearch = ""; // we'll get the input later.
```

Comment:
This will cause a bug. I want you to try and track down what the error would be because I think it is a good exercise in using/reading APIs. Check the API documentation and if something isn't clear, ask Andrew to clarify the comments.